src/ecalc/cli/tests/test_app.py

```diff
@@ -301,6 +301,28 @@ def test_json_true_detailed_output(self, simple_yaml_path, tmp_path, snapshot):
             json.dumps(json_data, sort_keys=True, indent=2, default=str), snapshot_name=v3_json_actual_path.name
         )
 
+    def test_json_advanced_model(self, advanced_yaml_path, tmp_path, snapshot):
+        run_name_prefix = "test_json_advanced_model"
+        runner.invoke(
+            main.app,
+            _get_args(
+                model_file=advanced_yaml_path,
+                json=True,
+                output_folder=tmp_path,
+                name_prefix=run_name_prefix,
+                detailed_output=True,
+            ),
+            catch_exceptions=False,
+        )
+
+        v3_json_actual_path = tmp_path / f"{run_name_prefix}_v3.json"
+        assert v3_json_actual_path.is_file()
+        with open(v3_json_actual_path) as json_file:
+            json_data = json.loads(json_file.read())
+        snapshot.assert_match(
```

Comment:
@TeeeJay, not sure why this fails in the pipeline. It succeeds when running locally.

Reply:
run `poetry install` locally, in case you have outdated dependencies installed locally

Reply:
Ran `poetry install `locally, didn´t help

Reply:
still fails only in pipeline? locally ok? do you run pytest in docker or not?

Reply:
Seems to be a classic difference between python (or numpy) versions. Which python version do you use when testing? the difference is in xth decimal AFAI can see, so no big problem. Ideally we should round off that output to fewer decimals, if possible, but I guess it is a dump, and hence not rounded off in serialization? Maybe we can do that after all? ...

Reply:
We use 3.8/3.9 in CICD, in general etc due to internal restrictions. To make sure there are no differences, I recommend to downgrade/use 3.9.17 (or newer) locally.

Reply:
this is why one should use docker, in order to make sure that the environment is correctly reproduced, and not dependent on  local configuration

Reply:
Tested with docker, python 3.8.14, 3.8.16, 3.8.18, 3.9.17 - same. Will try to modify the test, do it without snapshot.

Reply:
Same results in all versions? I can give it a go on my machine...possibly numpy version differs then, but that should be "locked"...